pkg/logpoller/query_test.go

```diff
@@ -0,0 +1,595 @@
+	store := &mockLogStore{}
+	addr, err := address.ParseAddr("EQDKbjIcfM6ezt8KjKJJLshZJJSqX7XOA4ff-W72r5gqPrHF")
+	require.NoError(t, err)
+
+	// Test saving logs
+	log1 := createTestLog(t, addr, 123, 42)
+	log2 := createTestLog(t, addr, 456, 24)
+	log3 := createTestLog(t, addr, 123, 84)
+
+	store.SaveLog(log1)
+	store.SaveLog(log2)
+	store.SaveLog(log3)
+
+	// Test retrieving logs by sig
+	logs123, err := store.GetLogs(addr, 123)
+	require.NoError(t, err)
+	require.Len(t, logs123, 2)
+
+	logs456, err := store.GetLogs(addr, 456)
+	require.NoError(t, err)
+	require.Len(t, logs456, 1)
+
+	// Test retrieving logs for non-existent sig
+	logsNone, err := store.GetLogs(addr, 999)
+	require.NoError(t, err)
+	require.Empty(t, logsNone)
+}
+
+func TestMockLogStore_DifferentAddresses(t *testing.T) {
+	store := &mockLogStore{}
+	addr1, err := address.ParseAddr("EQDKbjIcfM6ezt8KjKJJLshZJJSqX7XOA4ff-W72r5gqPrHF")
+	require.NoError(t, err)
+	addr2, err := address.ParseAddr("EQCD39VS5jcptHL8vMjEXrzGaRcCVYto7HUn4bpAOg8xqB2N")
+	require.NoError(t, err)
+
+	// Save logs for different addresses with same sig
+	log1 := createTestLog(t, addr1, 123, 42)
+	log2 := createTestLog(t, addr2, 123, 24)
+
+	store.SaveLog(log1)
+	store.SaveLog(log2)
+
+	// Verify each address only gets its own logs
+	logs1, err := store.GetLogs(addr1, 123)
+	require.NoError(t, err)
+	require.Len(t, logs1, 1)
+	require.True(t, logs1[0].Address.Equals(addr1))
+
+	logs2, err := store.GetLogs(addr2, 123)
+	require.NoError(t, err)
+	require.Len(t, logs2, 1)
+	require.True(t, logs2[0].Address.Equals(addr2))
+}
+
+func TestQueryBuilder_ExecuteWithLargeDataset(t *testing.T) {
+	store := &mockLogStore{}
+	addr, err := address.ParseAddr("EQDKbjIcfM6ezt8KjKJJLshZJJSqX7XOA4ff-W72r5gqPrHF")
+	require.NoError(t, err)
+
+	// Create a large dataset
+	for i := 0; i < 100; i++ {
+		store.SaveLog(createTestLog(t, addr, 123, uint64(i))) //nolint:gosec // test code
+	}
+
+	// Test basic query returns all data
+	builder := NewQuery[TestEvent](store).
+		WithSrcAddress(addr).
+		WithEventSig(123)
+
+	result, err := builder.Execute(context.Background())
+	require.NoError(t, err)
+	require.Len(t, result.Logs, 100)
+	require.Equal(t, 100, result.Total)
+	require.False(t, result.HasMore)
+}
+
+func TestQueryBuilder_ExecuteWithComplexFiltering(t *testing.T) {
+	store := &mockLogStore{}
+	addr, err := address.ParseAddr("EQDKbjIcfM6ezt8KjKJJLshZJJSqX7XOA4ff-W72r5gqPrHF")
+	require.NoError(t, err)
+
+	// Create test data with various values
+	testValues := []uint64{5, 10, 15, 20, 25, 30, 35, 40, 45, 50}
+	for _, val := range testValues {
+		store.SaveLog(createTestLog(t, addr, 123, val))
+	}
+
+	// Test typed filter: values > 20 and < 40
+	builder := NewQuery[TestEvent](store).
+		WithSrcAddress(addr).
+		WithEventSig(123).
+		WithTypedFilter(func(event TestEvent) bool {
+			return event.Value > 20 && event.Value < 40
+		})
+
+	result, err := builder.Execute(context.Background())
```

Comment:
Is the store ever used before the `Execute()` fn? Why not making it an arg of the Execute fn? I'm trying to figure out why the store dependency is required to create a query, it should be required only to execute a query.